### PR TITLE
docs: add new rubygems supply chain security story

### LINKED ISSUE #980: New supply chain security compromise issue that we should add to the list
RubyGems security flaw allow overriding gems: https://www.bleepingcomputer.com/news/security/check-your-gems-rubygems-fixes-unauthorized-package-takeover-bug/

### PR DESCRIPTION
**Pull request by lirantal**
Fixes #980 

Signed-off-by: Liran Tal <liran.tal@gmail.com>

### CODE REVIEW

**Comment by mnm678** [comment]
Thanks for submitting this! Could you fix the typos tagged by the linter and add the compromise to the [index](https://github.com/cncf/tag-security/tree/main/supply-chain-security/compromises)?

**Comment by lirantal** [comment]
@mnm678 fixed typos, rebased, and added DCO.

File: supply-chain-security/compromises/2022/rubygems-override.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# RubyGems Package Overwrite Flaw
+
+This security bug has been identified as CVE-2022-29176 and
+referring to the RubyGems.org website which is the home of
```
Comment (szh) [inline supply-chain-security/compromises/2022/rubygems-override.md:4]:
```suggestion
affects the RubyGems.org website which is the home of
```

File: supply-chain-security/compromises/2022/rubygems-override.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+open source libraries for the Ruby developer ecosystem.
+
+## Impact
+
+A security bug allowed anyone to remove legitimate packages
+on the RubyGems open source registry for Ruby, and then republish
+the same packages as new ones, with modified code, and potentially
```
Comment (szh) [inline supply-chain-security/compromises/2022/rubygems-override.md:11]:
```suggestion
the same packages as new, with modified code and potentially
```

File: supply-chain-security/compromises/2022/rubygems-override.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+malicious contents.
+
+Even though exploiting this vulnerability requires specific conditions
+to apply, such as a gem has to be of specific naming convention, its
+impact could still be devastating if went unnoticed.
```
Comment (szh) [inline supply-chain-security/compromises/2022/rubygems-override.md:16]:
```suggestion
impact could still be devastating if it were to go unnoticed.
```

File: supply-chain-security/compromises/2022/rubygems-override.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+As of yet, the RubyGems.org website maintainers haven't confirmed any
+known case of malicious exploitation they are aware of.
```
Comment (szh) [inline supply-chain-security/compromises/2022/rubygems-override.md:19]:
```suggestion
known cases of malicious exploitation they are aware of.
```

**Review by szh — CHANGES_REQUESTED** [review]
Some grammar nit picks :)

**Comment by lirantal** [comment]
Ready to land ✅

**Review by szh — APPROVED** [review]
LGTM! Will need @lumjjb, @PushkarJ or someone else with write access to merge.

**Comment by szh** [comment]
Any reason this hasn't been merged yet?

**Comment by lirantal** [comment]
I'll update the index entry, missed that!

**Comment by lirantal** [comment]
@szh @mnm678 ready now with the index up to date.

**Comment by lirantal** [comment]
Apologies on the pings. I had a mess with this branch and had to start over. PR changes moved to this new one: https://github.com/cncf/tag-security/pull/1078

**Pull request closed without merging**